### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by raihan2bd**
# Vet Clinic Database 🐻 
hello
### Hi there 👋 In this PR:

- [ ] I created a new table called vets ✅ 
- [ ] Then I created two junction tables/linking tables to the database ✅ 
- [ ] Then inserted Microverse given data to those tables ✅ 
- [ ] Wrote queries to answer the following question: ⏬ 
     - Who was the last animal seen by William Tatcher? ✔️ 
      q1
     - How many different animals did Stephanie Mendez see? ✔️ 
      q2
     - List all vets and their specialties, including vets with no specialties. ✔️ 
     q3
     - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. ✔️ 
     q4
     - What animal has the most visits to vets? ✔️ 
     q5
     - Who was Maisy Smith's first visit? ✔️ 
     q6
     - Details for most recent visit: animal information, vet information, and date of visit. ✔️ 
     q7
     - How many visits were with a vet that did not specialize in that animal's species? ✔️ 
     q8
     - What specialty should Maisy Smith consider getting? Look for the species she gets the most. ✔️ 
     q9

***If you have any questions feel free to ask ❓ Thanks for visiting my PR 👍***

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -33,4 +33,27 @@ ALTER TABLE animals
 ALTER TABLE animals 
     ADD foreign key (species_id) references species(id);
 ALTER TABLE animals 
-    ADD foreign key (owner_id) references owners(id);
\ No newline at end of file
+    ADD foreign key (owner_id) references owners(id);
+
+-- create vets table to the database
+CREATE TABLE vets (
+    id bigserial PRIMARY KEY,
```
Comment (t-yanick) [inline schema.sql]: [ OPTIONAL] 
- Great job working on this milestone. You can consider using `int` datatype. `bigserial` will occupy more memory just for storing primary key id

File: schema.sql [hunk 2]
```diff
@@ -33,4 +33,27 @@ ALTER TABLE animals
...
+    name VARCHAR(100),
+    age INT,
+    date_of_graduation DATE
+);
+
+-- create specializations join table to the database
+CREATE TABLE specializations (
+    id bigserial PRIMARY KEY,
```
Comment (t-yanick) [inline schema.sql]: - [ ] `specializations` table references the primary key of `species` and `vets` table. So just the foreign keys are necessary. Kindly delete the `id` `PRIMARY KEY` .

File: schema.sql [hunk 3]
```diff
@@ -33,4 +33,27 @@ ALTER TABLE animals
...
+    species_id INT REFERENCES species(id),
+    vet_id INT REFERENCES vets(id)
+);
+
+-- create visits table to join animals table and vets table
+CREATE TABLE visits (
+    id bigserial PRIMARY KEY,
```
Comment (t-yanick) [inline schema.sql]: - [ ] `visits` table references the primary key of `animals` and `vets` table. So just the foreign keys are necessary. Kindly delete the `id PRIMARY KEY` .

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @raihan2bd , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All SQL commands put in correct files
:heavy_check_mark: Screenshots of all queries added to PR description

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @raihan2bd, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally
✔️ All SQL commands put in correct files

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by raihan2bd** [comment]
Hi @Ismailco, @t-yanick ✋ Thanks a lot for you reviewing my PR. 👍

**Pull request merged**